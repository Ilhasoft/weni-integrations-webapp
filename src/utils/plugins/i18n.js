import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
