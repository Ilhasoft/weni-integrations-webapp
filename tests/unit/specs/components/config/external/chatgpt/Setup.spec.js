import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import chatGptSetup from '@/components/config/external/chatgpt/Setup.vue';
import i18n from '@/utils/plugins/i18n';
import { singleApp } from '../../../../../../__mocks__/appMock';
import { unnnicRadio } from '@weni/unnnic-system';

singleApp.code = 'chatgpt';

const localVue = createLocalVue();
localVue.use(Vuex);

const mountComponent = async ({ errorCreateApp = false } = {}) => {
  const state = {
    auth: {
      project: '123',
    },
    appType: {
      errorCreateApp,
      loadingCreateApp: false,
    },
  };

  const actions = {
    createApp: jest.fn(),
  };

  const store = new Vuex.Store({
    actions,
    state,
  });

  const wrapper = shallowMount(chatGptSetup, {
    localVue,
    i18n,
    store,
    propsData: {
      app: singleApp,
    },
    mocks: {
      $router: {
        replace: jest.fn(),
      },
      $route: {
        path: '/apps/1/details',
      },
    },
    stubs: {
      UnnnicModal: true,
      UnnnicButton: true,
      UnnnicIconSvg: true,
      UnnnicTooltip: true,
      unnnicRadio: true,
    },
  });

  return { wrapper, actions, state };
};

describe('components/config/external/chatgpt/Setup.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call parent closePopUp()', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper.emitted('closePopUp')).toBeFalsy();
    wrapper.vm.closePopUp();
    expect(wrapper.emitted('closePopUp')).toBeTruthy();
  });

  describe('setupChatGptService()', () => {
    it('should call createApp with the correct payload', async () => {
      const { wrapper, actions } = await mountComponent();

      const nameInput = wrapper.find('.chatgpt-modal__content__form__input__name');
      const tokenInput = wrapper.find('.chatgpt-modal__content__form__input__token');
      const versionRadio = wrapper
        .find('.chatgpt-modal__content__form__version-wrapper__options')
        .findAll(unnnicRadio)
        .at(0);

      nameInput.vm.$emit('input', 'chatgpt name');
      tokenInput.vm.$emit('input', 'chatgpt token');
      versionRadio.vm.$emit('change', 'gpt-3.5-turbo');

      expect(actions.createApp).not.toHaveBeenCalled();
      await wrapper.vm.setupChatGptService();
      expect(actions.createApp).toHaveBeenCalledWith(expect.anything(), {
        code: 'chatgpt',
        payload: {
          project_uuid: '123',
          name: 'chatgpt name',
          api_key: 'chatgpt token',
          ai_model: 'gpt-3.5-turbo',
        },
      });

      expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
        props: {
          text: 'Successfully created ChatGPT integration',
          title: 'Success',
          icon: 'check-circle-1-1',
          scheme: 'feedback-green',
          closeText: 'Close',
          position: 'bottom-right',
        },
        seconds: 6,
      });
    });

    it('should call unnnicCallAlert with error state', async () => {
      const { wrapper } = await mountComponent({ errorCreateApp: true });

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

      await wrapper.vm.setupChatGptService();

      expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
        props: {
          text: 'Failed to create ChatGPT integration, please try again later',
          title: 'Error',
          icon: 'alert-circle-1',
          scheme: 'feedback-red',
          closeText: 'Close',
          position: 'bottom-right',
        },
        seconds: 6,
      });
    });
  });

  describe('callModal', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('should call unnnicCallAlert with Success state', async () => {
      const { wrapper } = await mountComponent();

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

      wrapper.vm.callModal({ text: 'success', type: 'Success' });

      expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
        props: {
          text: 'success',
          title: 'Success',
          icon: 'check-circle-1-1',
          scheme: 'feedback-green',
          closeText: 'Close',
          position: 'bottom-right',
        },
        seconds: 6,
      });
    });

    it('should call unnnicCallAlert with Error state', async () => {
      const { wrapper } = await mountComponent();

      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

      wrapper.vm.callModal({ text: 'error', type: 'Error' });

      expect(mockUnnnicCallAlert).toHaveBeenCalledWith({
        props: {
          text: 'error',
          title: 'Error',
          icon: 'alert-circle-1',
          scheme: 'feedback-red',
          closeText: 'Close',
          position: 'bottom-right',
        },
        seconds: 6,
      });
    });
  });
});
