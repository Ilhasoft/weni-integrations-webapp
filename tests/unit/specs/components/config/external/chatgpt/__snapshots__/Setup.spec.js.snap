// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/config/external/chatgpt/Setup.vue should be rendered properly 1`] = `
<unnnicmodal-stub showmodal="true" class="chatgpt-modal">
  <div class="chatgpt-modal__content"><span class="chatgpt-modal__content__title">Integrate with ChatGPT</span> <span class="chatgpt-modal__content__description">To integrate, create a name, insert your API-Token and select one of the available versions</span>
    <div class="chatgpt-modal__content__form">
      <unnnic-input-stub placeholder="Insert a name" type="normal" value="" nativetype="text" label="Name" size="md" mask="" class="chatgpt-modal__content__form__input__name"></unnnic-input-stub>
      <unnnic-input-stub placeholder="Insert the Token" type="normal" value="" nativetype="text" label="Token" size="md" mask="" class="chatgpt-modal__content__form__input__token"></unnnic-input-stub>
      <div class="chatgpt-modal__content__form__version-wrapper">
        <div>
          Version

          <unnnic-tooltip-stub text="Changing the version of ChatGPT can change the way with that the AI ​​generates the answers based on the knowledge base and rules." enabled="true" side="right" maxwidth="350px" class="chatgpt-modal__content__form__version__tooltip">
            <unnniciconsvg-stub icon="information-circle-4" size="sm" scheme="neutral-soft" class="chatgpt-modal__content__form__version__icon"></unnniciconsvg-stub>
          </unnnic-tooltip-stub>
        </div>
        <div class="chatgpt-modal__content__form__version-wrapper__options">
          <unnnicradio-stub value="gpt-3.5-turbo-16k" globalvalue="gpt-3.5-turbo-16k" size="md" options="gpt-3.5-turbo-16k,gpt-3.5-turbo,gpt-4" label="Version">
            gpt-3.5-turbo-16k
          </unnnicradio-stub>
          <unnnicradio-stub value="gpt-3.5-turbo" globalvalue="gpt-3.5-turbo-16k" size="md" options="gpt-3.5-turbo-16k,gpt-3.5-turbo,gpt-4" label="Version">
            gpt-3.5-turbo
          </unnnicradio-stub>
          <unnnicradio-stub value="gpt-4" globalvalue="gpt-3.5-turbo-16k" size="md" options="gpt-3.5-turbo-16k,gpt-3.5-turbo,gpt-4" label="Version">
            gpt-4
          </unnnicradio-stub>
        </div>
      </div>
    </div>
  </div>
  <unnnicbutton-stub size="large" type="terciary">
    Cancel
  </unnnicbutton-stub>
  <unnnicbutton-stub size="large" type="secondary">
    Continue
  </unnnicbutton-stub>
</unnnicmodal-stub>
`;
